src/pkg/services/m365/api/item_pager.go

```diff
@@ -186,12 +276,37 @@ func deltaEnumerateItems[T any](
 
 	logger.Ctx(ctx).Debugw("completed delta item enumeration", "result_count", len(result))
 
-	du := DeltaUpdate{
+	npr.du = DeltaUpdate{
 		URL:   newDeltaLink,
 		Reset: invalidPrevDelta,
 	}
+}
+
+func batchDeltaEnumerateItems[T any](
+	ctx context.Context,
+	pager DeltaPager[T],
+	prevDeltaLink string,
+) ([]T, DeltaUpdate, error) {
+	var (
+		npr = nextPageResults[T]{
+			pages: make(chan nextPage[T]),
+		}
+		results = []T{}
+	)
+
+	go deltaEnumerateItems[T](ctx, pager, &npr, prevDeltaLink)
+
+	for is, reset, done := npr.NextPage(); !done; {
+		if reset {
+			results = []T{}
+		}
```

Comment:
just calling out that whatever behavior is decided for `NextPage` this should be updated to match (if needed)

Reply:
that's already compliant with expectations.